### PR TITLE
Change from a custom manylinux3 to prebuild and upstream manylinux2014

### PR DESCRIPTION
**Pull request by JanuszL**
- when cuda 9 is deprecated we can move to manylinux2014 as we can use the newer compiler than gcc 5.x
- CUDA 10 still required gcc 7 at most so we need to stick to it for all CUDA versions

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It changes from a custom manylinux3 to prebuild and upstream  manylinux2014

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     when cuda 9 is deprecated we can move to manylinux2014 as we can use the newer compiler than gcc 5.x
     CUDA 10 still required gcc 7 at most so we need to stick to it for all CUDA versions
 - Affected modules and functionalities:
     build system
 - Key points relevant for the review:
     NA
 - Validation and testing:
     local test, CI
 - Documentation (including examples):
     updated build instruction

In response to https://github.com/NVIDIA/DALI/issues/1957

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: docker/Dockerfile.deps [hunk 1]
```diff
@@ -160,7 +160,18 @@ RUN LIBSND_VERSION=1.0.28 && cd /tmp
...
-    cd /tmp && rm -rf libsnd-$LIBSND_VERSION
+    cd /tmp && rm -rf libsndfile-$LIBSND_VERSION
 
 # CUDA
 COPY --from=cuda /usr/local/cuda /usr/local/cuda
+
+# add devtoolset-7 to paths, it won't break anyting even if devtoolset-7 are not installed
+ENV ADDITIONAL_DEVTOOLSET_ROOTPATH /opt/rh/devtoolset-7/root
+ENV PATH $ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/bin:$PATH
+ENV LD_LIBRARY_PATH $ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib64:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib64/dyninst:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib/dyninst:$LD_LIBRARY_PATH
```
Comment (szalpal) [inline docker/Dockerfile.deps:171]:
```suggestion
# add devtoolset-7 to paths, it won't break anyting even if devtoolset-7 are not installed
ENV ADDITIONAL_DEVTOOLSET_ROOTPATH=/opt/rh/devtoolset-7/root    \
    PATH=$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/bin:$PATH         \
    LD_LIBRARY_PATH=$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib64:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib64/dyninst:$ADDITIONAL_DEVTOOLSET_ROOTPATH/usr/lib/dyninst:$LD_LIBRARY_PATH
```
Thanks to this we'll have only one additional layer in docker

Comment (JanuszL) [reply docker/Dockerfile.deps]: Done

Comment (szalpal) [reply docker/Dockerfile.deps]: You missed the `=` char ;)
```
[...]ROOTPATH=/opt/rh[...]
             ^ this one
```
Linking ref to docs: https://docs.docker.com/engine/reference/builder/#env

Comment (JanuszL) [reply docker/Dockerfile.deps]: Done

**Comment by JanuszL** [comment]
> LGTM, only comment is that maybe the docs in `Dockerfile.deps:1` would use some update

Done

File: dali/kernels/erase/erase_cpu.h [hunk 2]
```diff
@@ -111,7 +127,7 @@ class EraseCpu {
     const T *in_ptr = in.data;
     T *out_ptr = out.data;
     if (out_ptr != in_ptr) {
-      std::memcpy(out_ptr, in_ptr, volume(shape) * sizeof(T));
+      detail::CopyImpl(out_ptr, in_ptr, volume(shape) * sizeof(T));
```
Comment (klecki) [inline dali/kernels/erase/erase_cpu.h:130]: Isn't this error for anything with `sizeof(T) > 1`?

File: dali/kernels/erase/erase_cpu.h [hunk 3]
```diff
@@ -30,6 +30,22 @@ namespace kernels {
...
+template <typename O>
+inline std::enable_if_t<std::is_trivially_copyable<O>::value, void>
+CopyImpl(O *out, const O *in, size_t num) {
+  std::memcpy(out, in, num);
+}
+
+template <typename O>
+inline std::enable_if_t<!std::is_trivially_copyable<O>::value, void>
+CopyImpl(O *out, const O *in, size_t num) {
+  for (size_t i = 0; i < num; ++i) {
```
Comment (klecki) [inline dali/kernels/erase/erase_cpu.h:42]: With the usage below, you will go out of bounds as `num` is `count * sizeof(O)`

File: dali/kernels/erase/erase_cpu.h (hunk 2 above)
Comment (JanuszL) [reply dali/kernels/erase/erase_cpu.h:130]: ... ,right

File: dali/kernels/erase/erase_cpu.h (hunk 3 above)
Comment (JanuszL) [reply dali/kernels/erase/erase_cpu.h:42]: Fixed

**Pull request merged**